proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -1700,6 +1700,45 @@ public static void ctor(out S @this, ref int f)
 }
 ```
 
+#### Infer *safe-to-escape* of declaration expressions
+We adjust the span-safety proposal to automatically infer a *safe-to-escape* of a declaration expression like the following.
+    
+```cs
+ref struct RS
+{
+    public RS(ref int x) { } // assumed to be able to capture 'x'
+
+    static void M0(RS input, out RS output) => output = input;
+
+    static void M1()
+    {
+        var i = 0;
+        var rs1 = new RS(ref i); // safe-to-escape of 'rs1' is CurrentMethod
+        M0(rs1, out var rs2); // safe-to-escape of 'rs2' is CurrentMethod
+    }
+
+    static void M2(RS rs1)
+    {
+        M0(rs1, out var rs2); // safe-to-escape of 'rs2' is CallingMethod
+    }
+
+    static void M3(RS rs1)
+    {
+        M0(rs1, out scoped var rs2); // 'scoped' modifier forces safe-to-escape of 'rs2' to the current local scope (CurrentMethod or narrower).
+    }
+}
+
+```
+
+The *safe-to-escape* of a declaration variable from an `out` argument or deconstruction (`(var x, var y) = M()`) is the *narrowest* of the following:
+* calling method
+* if out variable is marked `scoped`, then the current local scope (i.e. current method or narrower).
+* if out variable's type is ref struct, consider all arguments to the containing invocation, including the receiver:
+  * STE of any argument where its corresponding parameter is not `out` and has STE of ReturnOnly or wider
+  * RSTE of any argument where its corresponding parameter has RSTE of ReturnOnly or wider
+    
```

Comment:
This change is all in the remarks while this is really a spec change. Would move this up into the actual spec section. Think we should add a section for declarations right below where we talk about the changes to `out` parameters.

Reply:
Done, thanks!